src/main/java/com/graphhopper/converter/resources/AbstractConverterResource.java

```diff
@@ -32,4 +34,31 @@ void checkInvalidParameter(boolean reverse, String query, String point) {
         }
     }
 
+    String getLocaleFromParameter(String locale){
+        Locale lo = parseLocale(locale);
+        if (isValid(lo)) {
+            return lo.toLanguageTag();
+        } else {
+            return "en";
+        }
+    }
+
+    // Taken from: http://stackoverflow.com/a/3684832/1548788
+    Locale parseLocale(String locale) {
+        String[] parts = locale.split("_");
```

Comment:
Thanks for the comment. I think I found an even better way to do this. There is a function in the Locale class to do this: https://docs.oracle.com/javase/7/docs/api/java/util/Locale.html#forLanguageTag(java.lang.String)

I added some more tests for it